buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/NamingConventionsTask.groovy

```diff
@@ -66,7 +65,7 @@ public class NamingConventionsTask extends LoggedExec {
     @Input
     boolean checkForTestsInMain = false;
 
-    public NamingConventionsTask() {
+    NamingConventionsTask() {
```

Comment:
Please add back the `public`. As I mentioned before, a goal here is to keep these files as close as possible to java so buildSrc can be converted completely to java (hopefully someday soon).

Reply:
Kotlin was not around when we switched to gradle. We have tried sticking to java for multiple reasons:
* Expertise on Elasticsearch is usually in java
* I have seen numerous reports of double digit performance gains in configuration time from switching gradle plugins to java

Remember we are only talking about buildSrc here. The build.gradle files are still groovy. I am not averse to them being Kotlin, but I would want to see a significant advantage to that work, as we have now spent over 2 years with developers learning how to write groovy in gradle DSL.